src/ApiHandler.ts

```diff
@@ -86,6 +86,7 @@ export default class ApiHandler {
 				info,
 				events: [] as SantiziedEvent[],
 				success: defaultSuccess,
+				paysFee: false, // override to true if `system.ExtrinsicSuccess` event is present
```

Comment:
Actually, this is part of `DispatchInfo`, which emits as part of `system.ExtrinsicFailure` too. So you should pay a fee even if it fails. Just certain dispatchables have weight but don't charge a fee. So it's a property of the extrinsic, not its success or failure.